client/dashboard/components/settings/general/store-address.tsx

```diff
@@ -130,98 +189,101 @@ export function getCountryStateOptions() {
  * @param {Function} setValue Set value of the countryState input.
  * @return {Object} React component.
  */
-export function useGetCountryStateAutofill( options, countryState, setValue ) {
+export function useGetCountryStateAutofill(
+	options: Option[],
+	countryState: string,
+	setValue: ( key: string, value: string ) => void
+): JSX.Element {
 	const [ autofillCountry, setAutofillCountry ] = useState( '' );
 	const [ autofillState, setAutofillState ] = useState( '' );
 	const isAutofillChange: {
 		current: boolean;
 	} = useRef();
 
+	// Sync the autofill fields on first render and the countryState value changes.
 	useEffect( () => {
-		const option = options.find( ( opt ) => opt.key === countryState );
-		const labels = option ? option.label.split( /\u2013|\u2014|\-/ ) : [];
-		const newCountry = ( labels[ 0 ] || '' ).trim();
-		const newState = ( labels[ 1 ] || '' ).trim();
+		if ( ! isAutofillChange.current ) {
+			const option = options.find( ( opt ) => opt.key === countryState );
+			const labels = option
+				? option.label.split( /\u2013|\u2014|\-/ )
+				: [];
+			const newCountry = ( labels[ 0 ] || '' ).trim();
+			const newState = ( labels[ 1 ] || '' ).trim();
 
-		if (
-			! isAutofillChange.current &&
-			( newCountry !== autofillCountry || newState !== autofillState )
-		) {
-			setAutofillCountry( newCountry );
-			setAutofillState( newState );
+			if (
+				newCountry !== autofillCountry ||
+				newState !== autofillState
+			) {
+				setAutofillCountry( newCountry );
+				setAutofillState( newState );
+			}
 		}
 		isAutofillChange.current = false;
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why is this needed? Is there a way to avoid this?

Reply:
I disabled this because we also update the `autofillCountry` and `autofillState` in this `useEffect`. If we include them in the dependency array, we will have an unnecessary function call whenever the `countryState` is changed.

We probably can remove these lines to avoid this, however, we then will have unnecessary state update calls.

```js
if (
  newCountry !== autofillCountry ||
  newState !== autofillState
) {
```